sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -547,15 +541,26 @@ puts that `command_graph` into an undefined state. Any further operations
 performed on a `command_graph` in this state will result in undefined
 behavior.
 
-==== Executable Graph Update
+===== Assume-Buffer-Outlives-Graph Property [[assume-buffer-outlives-graph-property]]
 
-A graph in the executable state can have each nodes inputs & outputs updated
-using the `command_graph::update()` method. This takes a graph in the
-modifiable state and updates the executable graph to use the node input &
-outputs of the modifiable graph, a technique called _Whole Graph Update_. The
-modifiable graph must have the same topology as the graph originally used to
-create the executable graphs, with the nodes targeting the same devices and
-added in the same order.
+The `property::graph::assume_buffer_outlives_graph` property disables
+<<buffer-limitations, restrictions on using buffers>> in a `command_graph` and
+can be passed to a `command_graph` on construction via the property list
+parameter. This property represents a promise from the user that any buffer
+which is used in a graph will be kept alive on the host for the lifetime of the
+graph. Destroying that buffer during the lifetime of a `command_graph`
+constructed with this property results in undefined behavior.
+
+===== Assume-Data-Outlives-Buffer Property [[assume-data-outlives-buffer-property]]
+
+The `property::graph::assume_data_outlives_buffer` property disables
+<<buffer-limitations, restrictions on using buffers>> which have been created
+with a host pointer in a `command_graph`, and can be passed to a `command_graph`
+on construction via the property list parameter. This property represents a
+promise from the user that any host data passed to a buffer's constructor will
+outlive the buffer itself, and by extension any graph in which that buffer is
+used. Deleting or otherwise modifying this host data during the lifetime of the
+buffer or graph results in undefined behavior when using this property.
```

Comment:
The core SYCL spec already requires the application to promise that the host pointer outlives the buffer.  This property represents an additional promise that the host pointer outlives the graph that records the buffer.  Therefore, you should remove the parts of the description talking about the lifetime of the host pointer outliving the buffer.  It's confusing to say this because that should already be the case even when not using this property.

```suggestion
The `property::graph::assume_data_outlives_buffer` property disables
<<buffer-limitations, restrictions on using buffers>> which have been created
with a host pointer in a `command_graph`, and can be passed to a `command_graph`
on construction via the property list parameter. This property represents a
promise from the user that any host data passed to a buffer's constructor will
outlive any graph in which that buffer is
used. Deleting or otherwise modifying this host data during the lifetime of the
graph results in undefined behavior when using this property.
```

Also, the `assume_data_outlives_buffer` isn't really necessary anymore since you added `assume_buffer_outlives_graph`.  If the user asserts that the buffer outlives the graph, normal SYCL rules require the host data to outlive the buffer.  Therefore, the host data will also outlive the graph.

I think I would suggest removing `assume_data_outlives_buffer` until you implement the buffer lifetime extension feature.  Once you do that, you can remove `assume_buffer_outlives_graph` and add `assume_data_outlives_buffer`.

Reply:
> I think I would suggest removing assume_data_outlives_buffer until you implement the buffer lifetime extension feature. Once you do that, you can remove assume_buffer_outlives_graph and add assume_data_outlives_buffer

This sounds like a good plan, removed `assume_data_outlives_buffer`